mlir/test/Dialect/Vector/canonicalize.mlir

```diff
@@ -2006,41 +2006,65 @@ func.func @shuffle_1d() -> vector<4xi32> {
   return %shuffle : vector<4xi32>
 }
 
+// -----
+
```

Comment:
[nit] Could you add a comment explaining what happens for poison idxs? It's quite non-obvious 😅🙏🏻